pkg/s3wrapper/client.go

```diff
@@ -33,23 +33,28 @@ const (
 	awsEndpointSuffix = ".amazonaws.com"
 )
 
+type ObjectInfo struct {
+	Path     string
+	Metadata map[string]*string
+}
+
 //go:generate mockgen --build_flags=--mod=mod -package=s3wrapper -destination=mock_s3wrapper.go . API
 //go:generate mockgen --build_flags=--mod=mod -package s3wrapper -destination mock_s3iface.go github.com/aws/aws-sdk-go/service/s3/s3iface S3API
 //go:generate mockgen --build_flags=--mod=mod -package s3wrapper -destination mock_s3manageriface.go github.com/aws/aws-sdk-go/service/s3/s3manager/s3manageriface UploaderAPI
 type API interface {
 	IsAwsS3() bool
 	CreateBucket() error
-	Upload(ctx context.Context, data []byte, objectName string) error
-	UploadStream(ctx context.Context, reader io.Reader, objectName string) error
-	UploadFile(ctx context.Context, filePath, objectName string) error
+	Upload(ctx context.Context, data []byte, objectName string, metadata map[string]*string) error
```

Comment:
Fair point, I think that we can do as you suggest and handle the conversion to and from a pointer within the S3 client layer, let me look into this.

Reply:
I have decided to use `map[string]string` everywhere instead of `map[string]*string` 

There are two helper methods in swag that take care of the conversion to and from these formats, 

`func StringMap(src map[string]string) map[string]*string`

and 

`func StringValueMap(src map[string]*string) map[string]string`

When calling `StringValueMap` a key value pair is deemed to be completely absent if the value is nil, this is adequate for our needs as we only want map[string]string in every case.

This only needs to be taken care of in the S3 client and allows us to maintain a clean interface as suggested.

This also addresses all the other concerns about absent values in `map[string]*string` so those comments will be marked as resolved.